### PR TITLE
Add exception for Arch in timezone zone_compare

### LINKED ISSUE #61296: [BUG] timezone.system fails on Archlinux if the path /etc/localtime is missing as it cannot determine the current localtime beforehand
**Description**

Using a timezone.system state on Archlinux fails with `Unable to compare desired timezone 'myCountry/myCity' to system timezone: Failed to read /etc/localtime to determine current timezone: No such file or directory`

This is similar to what used to happen on FreeBSD in https://github.com/saltstack/salt/issues/44083 however that was patched back in 2017, as referenced in that issue.

**Setup**
(Please provide relevant configs and/or SLS files (be sure to remove sensitive info. There is no general set-up of Salt.)

Please be as specific as possible and give set-up details.

- [x] on-prem machine
- [ ] VM (Virtualbox, KVM, etc. please specify)
- [ ] VM running on a cloud service, please be explicit and add details
- [ ] container (Kubernetes, Docker, containerd, etc. please specify)
- [ ] or a combination, please be explicit
- [ ] jails if it is FreeBSD

This is a kind of issue which will happen on any Archlinux install in any of these above environments.
  
**Steps to Reproduce the behavior**

1. Make a salt state with the timezone.system state while /etc/localtime does not exist
2. Run that state
3. Receive failed state with comment: `Unable to compare desired timezone 'myCountry/myCity' to system timezone: Failed to read /etc/localtime to determine current timezone: No such file or directory`

**Expected behavior**

I was hoping it would just create the symlink and move on.

**Screenshots**

**Versions Report**
<details><summary>salt --versions-report</summary>
(Provided by running salt --versions-report. Please also mention any differences in master/minion versions.) 

```
Salt Version:
          Salt: 3004
 
Dependency Versions:
          cffi: Not Installed
      cherrypy: Not Installed
      dateutil: Not Installed
     docker-py: Not Installed
         gitdb: Not Installed
     gitpython: Not Installed
        Jinja2: 3.0.3
       libgit2: Not Installed
      M2Crypto: 0.38.0
          Mako: Not Installed
       msgpack: 1.0.2
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     pycparser: Not Installed
      pycrypto: Not Installed
  pycryptodome: 3.10.1
        pygit2: Not Installed
        Python: 3.9.7 (default, Oct 10 2021, 15:13:22)
  python-gnupg: Not Installed
        PyYAML: 5.4.1
         PyZMQ: 22.2.1
         smmap: Not Installed
       timelib: Not Installed
       Tornado: 4.5.3
           ZMQ: 4.3.4
 
System Versions:
          dist: arch  
        locale: utf-8
       machine: x86_64
       release: 5.15.5-arch1-1
        system: Linux
       version: Arch Linux  
```
</details>

**Additional context**

Not entirely world ending, I've added a jinja if in my Salt to just run a fallback state to create a dummy symlink first before this state runs just so it doesn't throw an error and just updates the timezone but in a perfect world the state would just make the change even if /etc/localtime is ever missing.

The issue feels like salt wants to do a comparison to see if it actually needs to make the change, but if the symlink is missing "how could it".

### PR DESCRIPTION
**Pull request by ipaqmaster**
Resubmission of #61390

Like FreeBSD an Archlinux host may not have a /etc/localtime symlink yet which can cause a comparison error during provisioning. It would be nice to not have to work around this problem any more with this small addition.

_If tests are required I could use a hand writing a suitable one._

### What does this PR do?

Prevents `zone_compare` in `salt/modules/timezone.py` throwing an error when /etc/localtime is missing on Archlinux hosts.

### What issues does this PR fix or reference?
Fixes: #61296

### Previous Behavior

```
[ERROR   ] Unable to compare desired timezone 'Australia/Melbourne' to system timezone: Failed to read /etc/localtime to determine current timezone: No such file or directory
local:
----------
          ID: Australia/Melbourne
    Function: timezone.system
      Result: False
     Comment: Unable to compare desired timezone 'Australia/Melbourne' to system timezone: Failed to read /etc/localtime to determine current timezone: No such file or directory
     Started: 06:37:43.845836
    Duration: 0.653 ms
     Changes:   

Summary for local
------------
Succeeded: 0
Failed:    1
------------
Total states run:     1
Total run time:   0.653 ms
```

### New Behavior

```
local:
----------
          ID: Australia/Melbourne
    Function: timezone.system
      Result: True
     Comment: Set timezone Australia/Melbourne
     Started: 06:38:29.172314
    Duration: 78.068 ms
     Changes:   
              ----------
              timezone:
                  Australia/Melbourne

Summary for local
------------
Succeeded: 1 (changed=1)
Failed:    0
------------
Total states run:     1
Total run time:  78.068 ms
```

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes/No

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by ipaqmaster** [comment]
Been trying to get this officially fixed for a while, any help or guidance on how to make this happen would be greatly appreciated.

**Comment by raymonad** [comment]
`"Arch"` by itself is Truthy, so this is not an exception for Arch but a check for all (supported) OSes. I don't understand why a missing file should be treated differently on some OSes in the first place and I see this also came up in the referenced issue.

The special cases for AIX/Solaris are documented, but this one (whether for FreeBSD/Arch only or as a general check) is not. It could also be argued that, if you don't want to throw an exception, you should simply return False if /etc/localtime doesn't exist, because it can't possibly compare equal to any zone file. Then Salt (states.timezone) would always create /etc/localtime; now it won't if the desired timezone is equal to the system default (probably UTC).

**Comment by ipaqmaster** [comment]
You're right that the statement was Truthy. I've fixed this in a new push but I have not been able to write tests.

Is there any 'gotcha' on why this section cannot just be:
```
    if not os.path.isfile(_get_localtime_path()):
        return timezone == get_zone()
```

Without an OS check?

**Review by Ch3LL — CHANGES_REQUESTED** [review]
this will require a changelog and it should be pretty easy to add some test coverage.

There are already tests in tests/unit/modules/test_timezone.py, if you could just add a test that mocks grains to be arch or freebsd so we ensure we are testing this code path as well. thanks

**Comment by ipaqmaster** [comment]
Thank you Ch3LL, I'm in a bit of a pickle at the moment with time management but I'll try and get this done

**Comment by Ch3LL** [comment]
k, i'm more than willing to help write the tests, so let me know if you can't find the time.

**Comment by ipaqmaster** [comment]
Sorry I won't be able to take care of this one at the current time. Despite how badly I'm itching to get this problem fixed.

File: salt/modules/timezone.py [hunk 1]
```diff
@@ -350,7 +350,7 @@ def zone_compare(timezone):
     if "Solaris" in __grains__["os_family"] or "AIX" in __grains__["os_family"]:
         return timezone == get_zone()
 
-    if "FreeBSD" in __grains__["os_family"]:
+    if __grains__["os_family"] in {"Arch", "FreeBSD"}:
```
Comment (waynew) [inline salt/modules/timezone.py:353]: **suggestion**: FWIW a tuple `(...)` is typically preferred over a set `{...}` for this kind of comparison:

```suggestion
    if __grains__["os_family"] in ("Arch", "FreeBSD"):
```

though looking at the code around it, would it be more appropriate to do

```
if "Arch" in __grains__["os_family"] or "FreeBSD" in __grains__["os_family"]:
```
? It seems like at least the other checks are thinking that `os_family` could be a list-type, or that they might contain more than just say "FreeBSD" but might have something like "Cool FreeBSD" or something to that effect.

**Review by waynew — CHANGES_REQUESTED** [review]
Yeah, this definitely does need changes, I also had a suggestion and a question.

Tests should be pretty easy to write - if you need some help feel free to come by one of the [Test Clinics on Twitch](https://www.twitch.tv/saltprojectoss/schedule).

**Comment by waynew** [comment]
@ipaqmaster are you interested in adding the tests and getting this merged in?

**Comment by Ch3LL** [comment]
He previously stated he doesn't have the time. I'm willing to add some tests coverage here unless you want to tackle this in your test clinic @waynew .

Just let me know

**Comment by waynew** [comment]
@Ch3LL I probably won't be able to tackle this before mid Nov/Dec, so if you get to it before that, feel free to take it on!

**Comment by Ch3LL** [comment]
I went ahead and pushed some test coverage, and made the requested changes.

@waynew and @ipaqmaster can you give it a look and see if everything looks good.

**Review by garethgreenaway — APPROVED** [review]

**Pull request merged**